### ISSUE TITLE
hakyl 4.8.1.0 failing to built: src/Hakyll/Web/Tags.hs:100:17: No instance for (Show (String -> Identifier))

### ISSUE THREAD

**Issue by bacher09** [jaspervdj/hakyll#420]
Last hakyll failing to built with this error:

```
src/Hakyll/Web/Tags.hs:100:17:
    No instance for (Show (String -> Identifier))
      (maybe you haven't applied enough arguments to a function?)
      arising from the second field of ‘Tags’
        (type ‘String -> Identifier’)
    Possible fix:
      use a standalone 'deriving instance' declaration,
        so you can specify the instance context yourself
    When deriving the instance for (Show Tags)
cabal: Error: some packages failed to install:
```
Here's my ghc and cabal versions:

```
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.10.3
$ cabal --version
cabal-install version 1.22.6.0
using version 1.22.5.0 of the Cabal library
```

**Comment by jaspervdj** [comment]
This is bizarre! The code causing this is:

```haskell
data Tags = Tags
    { tagsMap        :: [(String, [Identifier])]
    , tagsMakeId     :: String -> Identifier
    , tagsDependency :: Dependency
    } deriving (Show)
```

and it should be indeed impossible to derive `Show`, given `tagsMakeId`... however it works on my machine. I wonder if a really weird orphan instance is causing this.

**Comment by jaspervdj** [comment]
Traced it down to `Text.Regex.TDFA` providing a `Show` instance for that function at some point. I'll remove the instance altogether, that should fix it.

**Comment by juhp** [comment]
cc @ChrisKuklewicz

**Comment by neongreen** [comment]
`import Text.Show.Functions` will fix this. The instance wasn't provided by regex-tdfa, it was accidentally reexported by regex-tdfa, which was a problem for some *other* users, and that's why I removed it.

Obviously I should've bumped it to 1.3.0 instead of 1.2.2 to comply with the PVP, and I apologise for accidentally breaking your code.

**Comment by jaspervdj** [comment]
Issue fixed in `hakyll-4.8.3.0`. :-)
